src/pages/Scholarships.jsx

```diff
@@ -21,11 +24,51 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 function ScholarshipsPage() {
+  const classes = useStyles();
   const [scholarships, setScholarships] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState();
-  const [sortField, setSortField] = useState('deadline');
-  const [sortDir, setSortDir] = useState('asc');
+  const history = useHistory();
+  const location = useLocation();
+
+  const validField = Boolean(
+    queryString.parse(location.search).sortField === 'deadline' ||
+      queryString.parse(location.search).sortField === 'amount.min' ||
+      queryString.parse(location.search).sortField === 'amount.max'
+  );
+  const validDir = Boolean(
+    queryString.parse(location.search).sortDir === 'asc' ||
+      queryString.parse(location.search).sortDir === 'desc'
+  );
+
+  let sortField = validField
+    ? queryString.parse(location.search).sortField
+    : DEFAULT_SORT_FIELD;
+  let sortDir = validDir
+    ? queryString.parse(location.search).sortDir
+    : DEFAULT_SORT_DIR;
+
+  const setSort = (val) => {
+    if (val === 'deadlineSoon') {
+      sortField = 'deadline';
+      sortDir = 'asc';
+    } else if (val === 'deadlineLatest') {
+      sortField = 'deadline';
+      sortDir = 'desc';
+    } else if (val === 'amountLow') {
+      sortField = 'amount.min';
+      sortDir = 'asc';
+    } else if (val === 'amountHigh') {
+      sortField = 'amount.max';
+      sortDir = 'desc';
+    }
+    history.push({
+      search: queryString.stringify({
+        sortField,
+        sortDir,
+      }),
```

Comment:
This is actually stripping all the other query params besides sortField and sortDir. You can see this by adding a different query param in the URL (e.g. `foo=bar`) and watching it disappear when you select a sorting option.

Reply:
Ah ok so how can I work around not stripping all the other query params and still add sortField and sortDir to the url?

Reply:
I think I was able to fix it with the most current commit, can you take a look and confirm?

Reply:
This is still relevant btw.

Reply:
Ahhhh what will fix it then